intelmq/bots/outputs/amqptopic/output.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import pika
+import sys
+import time
+
+from intelmq.lib.bot import Bot
+
+
+class AMQPTopicBot(Bot):
+
+    CONTENT_TYPE = 'application/json'
+    DELIVERY_MODE = 1
+
+    def init(self):
+        self._connection = None
+        self._channel = None
+        self._url = self.parameters.url
+        self._retries = self.parameters.retries
+        self._maxerrors = self.parameters.maxerrors
+        self._delay = self.parameters.delay
+        self._exchange = self.parameters.exchange
+        self._routing_key = self.parameters.routingkey
+        self.connect_server()
+
+    def connect_server(self):
+        self.logger.info("Connecting to %s", self._url)
+        for i in range(0, self._retries):
+            try:
+                self._connection = pika.BlockingConnection(
+                    pika.URLParameters(self._url))
+                break
+            except pika.exceptions.AMQPConnectionError:
+                self.logger.info(
+                    'Could not connect to server. Retrying... (%d/%d)' % (i + 1, self._retries))
+                time.sleep(self._delay)
+            except Exception, e:
+                self.logger.exception('Unknown exception: %s', e)
```

Comment:
You may use the `traceback` module and it's functions `print_exc` or `format_exc` to give the error including traceback information.